.github/workflows/build-macos.yaml

```diff
@@ -0,0 +1,26 @@
+name: Build on Mac OS
+run-name: Build on Mac OS
+
+on:
+  pull_request:
```

Comment:
I've just realized that we probably should only be invoking these workflows if there are code changes, similar to how we only run unit and other tests based on path filters? I can make this changes as a part of the [buddy PR](https://github.com/gravitational/teleport/pull/19226).